src/content/docs/en/reference/adapter-reference.mdx

```diff
@@ -136,14 +136,32 @@ export function start(manifest) {
 
 The following methods are provided:
 
-##### `app.render(request)`
+##### `app.render(request, routeData, locals)`
 
 This method calls the Astro page that matches the request, renders it, and returns a Promise to a [Response](https://developer.mozilla.org/en-US/docs/Web/API/Response) object. This also works for API routes, that do not render pages.
 
 ```js
 const response = await app.render(request);
 ```
 
+The method accepts a mandatory `request` arguments, and two other optionals arguments: `routeData` and `locals`.
+
+[//]: # (TODO) document `routeData`
+
+`locals` is an optional object to pass to the adapter. This object will be passed to any [middleware function](/en/guides/middleware/).
```

Comment:
```suggestion
The example below stores a private header in the `locals` object, which can then be passed to any [middleware function](/en/guides/middleware/).
```
Just example wording (which you'll have to make correct!), but what do you think about something like this?